### PR TITLE
create Dice Game and added test module

### PR DESCRIPTION
**Pull request by this-is-shivamsingh**
Please review the code

### CODE REVIEW

File: helper.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+end
+
+def next_turn(currentPlayer, totalPlayer)
+    if(totalPlayer == 0)
+        puts "No of Players should be greater than 0"
+        return
+    end
+    currentPlayer = (currentPlayer + 1)%(totalPlayer)
+    return currentPlayer
+end
```
Comment (gotunandan) [inline helper.rb:110]: please add a new line at EOF

Comment (gotunandan) [reply helper.rb:110]: This is needed for all files below this as well

File: diceGame.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
+require_relative 'helper'
+require_relative 'userInput'
+require_relative 'player'
+
+$currentPlayerIdx = -1
+$no_of_player = 0
+$max_score = 3000
+$min_score = 300
+$no_of_dices = 5
+def set_user_score(currentPlayerIdx, totalScore)
```
Comment (gotunandan) [inline diceGame.rb:10]: Please use 2 space indentation instead of 4

Comment (gotunandan) [reply diceGame.rb:10]: You can follow this convention https://gist.github.com/gotunandan/f886e6bac00a0d3aa0512a41af7d7d20

File: diceGame.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    count.times do
+        players.push(Player.new(cnt))
+        cnt += 1
+    end
+    return players
+end
+
+def startGame
+    "Enter the no. of players: "
+    $no_of_player = gets
```
Comment (gotunandan) [inline diceGame.rb:102]: Not seeing this prompt
Also should check for valid inputs

```
root@bebec318b125:/rancher-desktop/greed_game# ruby index.rb
a
No of Players should be greater than 0
/rancher-desktop/greed_game/diceGame.rb:42:in `play_your_turn': undefined method `+' for nil:NilClass (NoMethodError)
	from /rancher-desktop/greed_game/diceGame.rb:110:in `startGame'
	from index.rb:2:in `<main>'
root@bebec318b125:/rancher-desktop/greed_game#
```

File: diceGame.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+require_relative 'player'
+
+$currentPlayerIdx = -1
+$no_of_player = 0
+$max_score = 3000
+$min_score = 300
+$no_of_dices = 5
+def set_user_score(currentPlayerIdx, totalScore)
+    currentScore = $players[currentPlayerIdx].totalScore
+    if currentScore == 0 && totalScore < $min_score
```
Comment (gotunandan) [inline diceGame.rb:12]: There is a bug here.

player 2 has scored 500, but it still does not consider it

```
Player 2 rolls: 1 3 3 3 1
Score in this turn: 500
Player 2 needs to have a score greater than 300 in single turn, before scores get acculumated
Player 2 TOTAL SCORE: 0
```

**Review by gotunandan — CHANGES_REQUESTED** [review]
Please make the requested changes

File: helper.rb (hunk 1 above)
Comment (this-is-shivamsingh) [reply helper.rb]: Resolved

File: diceGame.rb (hunk 4 above)
Comment (this-is-shivamsingh) [reply diceGame.rb:13]: Issue resides in,
play_your_turn function, where breaking the statement when non_scoring_dice == 0, but not updating totalScore .
Status: Resolved.

Comment (gotunandan) [reply diceGame.rb:13]: This Looks good

**Review by gotunandan — CHANGES_REQUESTED** [review]
The bug is fixed, but I can still see multiple places where 4 space indentation is being used instead of 2.
See line 20 21 and line 22 for example.

All functions have 4 space indentation for the first line

**Comment by this-is-shivamsingh** [comment]
Change indentation from 4 to 2 in latest commit. 
Do you also suggest any vs code plugin to do so like - preittier for js ?

**Review by gotunandan — APPROVED** [review]
Looks good to me

**Comment by this-is-shivamsingh** [comment]
PR approved, closing this PR

**Pull request closed without merging**